packages/react-native/ReactCommon/react/renderer/components/scrollview/ScrollViewEventEmitter.cpp

```diff
@@ -25,8 +25,9 @@ void ScrollViewEventEmitter::onScrollBeginDrag(
 }
 
 void ScrollViewEventEmitter::onScrollEndDrag(
-    const ScrollEvent& scrollEvent) const {
-  dispatchScrollViewEvent("scrollEndDrag", scrollEvent);
+    const ScrollEndDragEvent& scrollEvent) const {
+  dispatchEvent(
+      "scrollEndDrag", std::make_shared<ScrollEndDragEvent>(scrollEvent));
```

Comment:
I'm not sure we need to create a new pointer with `make_shared` here.
The type of the scrollEvent is already the right one. If previously we were passing the scrollEvent, we should be able to pass the scrollEvent even now, types should match.

Reply:
@cipolleschi Previously we call `dispatchScrollViewEvent ` but it std::make_shared<ScrollEvent> to create `ScrollEvent` shared ptr.

Reply:
`ScrollEndDragEvent` event is a subclass of `ScrollEvent`, we probably don't need to change this method at all, as you are allowed to pass a subclass to a method that accepts a superclass. The language should take care of it. ([LSP - Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle))

Can you remove these changes and test if this works properly without these?